apimodels/agent_models.go

```diff
@@ -114,8 +125,38 @@ type EbsDevice struct {
 	SnapshotID string `mapstructure:"ebs_snapshot_id" json:"ebs_snapshot_id"`
 }
 
-func (ch *CreateHost) Validate() error {
+type RegistrySettings struct {
+	Name     string `mapstructure:"registry_name" json:"registry_name"`
+	Username string `mapstructure:"registry_username" json:"registry_username"`
+	Password string `mapstructure:"registry_password" json:"registry_password"`
+}
+
+func (ch *CreateHost) ValidateDocker() error {
+	catcher := grip.NewBasicCatcher()
+	if ch.Distro == "" {
+		catcher.Add(errors.New("distro must be set"))
+	}
+	if ch.Image == "" || ch.Command == "" {
```

Comment:
These feel like they should be separate checks so that the error message is clearer. (It'll say exactly what wasn't set.)